hudi-common/src/test/java/org/apache/hudi/avro/TestMercifulJsonConverter.java

```diff
@@ -55,6 +70,649 @@ public void basicConversion() throws IOException {
     Assertions.assertEquals(rec, CONVERTER.convert(json, simpleSchema));
   }
 
+  private static final String DECIMAL_AVRO_FILE_INVALID_PATH = "/decimal-logical-type-invalid.avsc";
+  private static final String DECIMAL_AVRO_FILE_PATH = "/decimal-logical-type.avsc";
+  private static final String DECIMAL_FIXED_AVRO_FILE_PATH = "/decimal-logical-type-fixed-type.avsc";
+  /**
+   * Covered case:
+   * Avro Logical Type: Decimal
+   * Exhaustive unsupported input coverage.
+   */
+  @ParameterizedTest
+  @MethodSource("decimalBadCases")
+  void decimalLogicalTypeInvalidCaseTest(String avroFile, String strInput, Double numInput,
```

Comment:
Similar for the tests on code porting.  And do these cover all cases of logical types?

Reply:
Yes, @the-other-tim-brown has confirmed on the type of logical types that we support. Test achieves at least 95% of branch coverage

For the code porting, it is replied in the previous thread.